lib/beaker/options/validator.rb

```diff
@@ -0,0 +1,131 @@
+module Beaker
+  module Options
+
+    class Validator
+      VALID_FAIL_MODES              = /stop|fast|slow/
+      VALID_PRESERVE_HOSTS          = /always|onfail|onpass|never/
+      FRICTIONLESS_ROLE             = 'frictionless'
+      FRICTIONLESS_ADDITIONAL_ROLES = %w(master database dashboard console)
+
+      # Determine is a given file exists and is a valid YAML file
+      # @param [String] f The YAML file path to examine
+      # @param [String] msg An options message to report in case of error
+      # @raise [ArgumentError] Raise if file does not exist or is not valid YAML
+      def check_yaml_file(f, msg = '')
+        validator_error "#{f} does not exist (#{msg})" unless File.file?(f)
+
+        begin
+          YAML.load_file(f)
+        rescue Beaker::Options::Parser::PARSE_ERROR => e
+          validator_error "#{f} is not a valid YAML file (#{msg})\n\t#{e}"
+        end
+      end
+
+      # resolves all file symlinks that require it.
+      #
+      # @param [Beaker::OptionsHash] options Beaker Options hash
+      # @note doing it here allows us to not need duplicate logic, which we
+      #   would need if we were doing it in the parser (--hosts & --config)
+      #
+      # @return nil
+      # @api public
+      def resolve_symlinks(options)
```

Comment:
Only if the file path causes `File.realpath` to throw an error. I am going to put it back in `parser.rb`.